File: ruby/hackerrank/apple_and_orange.rb
#!/bin/ruby

require 'json'
require 'stringio'

#
# Complete the 'countApplesAndOranges' function below.
#
# The function accepts following parameters:
#  1. INTEGER s
#  2. INTEGER t
#  3. INTEGER a
#  4. INTEGER b
#  5. INTEGER_ARRAY apples
#  6. INTEGER_ARRAY oranges
#

def countApplesAndOranges(s, t, a, b, apples, oranges)
  apple_count = 0
  orange_count = 0

  apples.each do |apple|
    apple_count += 1 if (a + apple).between?(s, t)
  end

  oranges.each do |orange|
    orange_count += 1 if (b + orange).between?(s, t)
  end

  puts apple_count
  puts orange_count
end

first_multiple_input = gets.rstrip.split

s = first_multiple_input[0].to_i

t = first_multiple_input[1].to_i

second_multiple_input = gets.rstrip.split

a = second_multiple_input[0].to_i

b = second_multiple_input[1].to_i

third_multiple_input = gets.rstrip.split

m = third_multiple_input[0].to_i

n = third_multiple_input[1].to_i

apples = gets.rstrip.split.map(&:to_i)

oranges = gets.rstrip.split.map(&:to_i)

countApplesAndOranges s, t, a, b, apples, oranges

File: java/hackerrank/apple_and_orange.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;

enum Result {
    ;

    /*
     * Complete the 'countApplesAndOranges' function below.
     *
     * The function accepts following parameters:
     *  1. INTEGER s
     *  2. INTEGER t
     *  3. INTEGER a
     *  4. INTEGER b
     *  5. INTEGER_ARRAY apples
     *  6. INTEGER_ARRAY oranges
     */

    public static void countApplesAndOranges(
            final int s,
            final int t,
            final int a,
            final int b,
            final List<Integer> apples,
            final List<Integer> oranges) {
        final int numApple = apples.size();
        final int numOrange = oranges.size();
        int appleCount = 0;
        int orangeCount = 0;
        for (int apple = 0; apple < numApple; apple++) {
            final int appDist = a + apples.get(apple);
            if (appDist >= s && appDist <= t) {
                appleCount++;
            }
        }
        for (int orange = 0; orange < numOrange; orange++) {
            final int orangeDist = b + oranges.get(orange);
            if (orangeDist >= s && orangeDist <= t) {
                orangeCount++;
            }
        }

        System.out.printf("%d%n%d%n", appleCount, orangeCount);
    }
}

public class Solution {
    public static void main(final String[] args) throws IOException {
        final BufferedReader bufferedReader =
                new BufferedReader(
                        new InputStreamReader(System.in, java.nio.charset.StandardCharsets.UTF_8));

        final String[] firstMultipleInput =
                bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        final int s = Integer.parseInt(firstMultipleInput[0]);

        final int t = Integer.parseInt(firstMultipleInput[1]);

        final String[] secondMultipleInput =
                bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        final int a = Integer.parseInt(secondMultipleInput[0]);

        final int b = Integer.parseInt(secondMultipleInput[1]);

        final String[] thirdMultipleInput =
                bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

        final int m = Integer.parseInt(thirdMultipleInput[0]);

        final int n = Integer.parseInt(thirdMultipleInput[1]);

        final List<Integer> apples =
                Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList());

        final List<Integer> oranges =
                Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList());

        Result.countApplesAndOranges(s, t, a, b, apples, oranges);

        bufferedReader.close();
    }
}
